R/get_-functions.R

```diff
@@ -209,10 +209,12 @@ get_protected_columns <- function(data = NULL) {
 
   protected_columns <- c(
     "predicted", "observed", "sample_id", "quantile", "upper", "lower",
-    "pit_value", "range", "boundary", "relative_skill", "scaled_rel_skill",
+    "pit_value", "range", "boundary",
     "interval_coverage", "interval_coverage_deviation",
     "quantile_coverage", "quantile_coverage_deviation",
     available_metrics(),
+    paste(available_metrics(), "relative_skill", sep = "_"),
```

Comment:
do we want to do this in the same way as the coverage is detected (i.e. using `grep`) rather than `available_metics()`? This will have problems when custom metrics are used I think.

Reply:
I think we definitely want to to deprecate `available_metrics()`. Maybe using `get_score_names()` instead would work. 
A separate question I've been thinking about is whether the score names should be part of `protected_columns()` at all. I.e. we could have `protected_columns()` just return the internal columns such as "observed", "predicted", "quantile" etc. and handle score names using `get_score_names()` separately.

Reply:
Would it make sense to merge this as is and then deprecate `available_metrics()` separately? It seems a separate issue.